lib/spack/spack/stage.py

```diff
@@ -432,13 +430,9 @@ def fetch(self, mirror_only=False, err_msg=None):
             # If this archive is normally fetched from a tarball URL,
             # then use the same digest.  `spack mirror` ensures that
             # the checksum will be the same.
-            digest = None
```

Comment:
I'm not sure that we actually want this change? Might have been reverted by accident when switching away from CurlFetchStrategy?